src/dharma.ts

```diff
@@ -27,25 +41,27 @@ class Dharma {
 
     public readonly web3: Web3;
 
-    constructor(blockchainHost?: string, addressBook: DharmaTypes.AddressBook = {}) {
+    constructor(web3?: Web3, addressBook: DharmaTypes.AddressBook = {}) {
```

Comment:
there is a world where this web3 instance is not optional.

`Dharma.default()` -> instantiates Dharma by pulling web3 off the window.